src/demo-app/a11y/a11y.ts

```diff
@@ -43,8 +47,17 @@ export class AccessibilityDemo {
     {name: 'Tooltip', route: 'tooltip'},
   ];
 
+  constructor(private router: Router) {
+    router.events.subscribe(event => {
+      let nav = this.navItems.find(navItem => {
+        let fragments = (event as NavigationEnd).url.split('/');
+        return fragments[fragments.length - 1] === navItem.route;
+      });
+      this.currentComponent = nav ? nav.name : '';
+    });
+  }
+
   skipNavigation() {
-    this.mainContent.nativeElement.scrollIntoView();
-    this.mainContent.nativeElement.focus();
+    (this.currentComponent ? this.sectionHeader : this.mainContent).nativeElement.focus();
```

Comment:
Did you find that this worked well without a timeout? If so that works for me.

Reply:
This worked well without a timeout. I tested with chromeVox and VoiceOver.